crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1298 @@
+    /// :attr:`~qiskit.transpiler.Target.instructions` returned::
+    ///
+    ///     [(XGate(), (0,)), (XGate(), (1,))]
+    ///
+    /// you could get the properties of the ``XGate`` on qubit 1 with::
+    ///
+    ///     props = target.instruction_properties(1)
+    ///
+    /// but just accessing it directly via the name would be more efficient::
+    ///
+    ///     props = target['x'][(1,)]
+    ///
+    /// (assuming the ``XGate``'s canonical name in the target is ``'x'``)
+    /// This is especially true for larger targets as this will scale worse with the number
+    /// of instruction tuples in a target.
+    ///
+    /// Args:
+    ///     index (int): The index of the instruction tuple from the
+    ///         :attr:`~qiskit.transpiler.Target.instructions` attribute. For, example
+    ///         if you want the properties from the third element in
+    ///         :attr:`~qiskit.transpiler.Target.instructions` you would set this to be ``2``.
+    /// Returns:
+    ///     InstructionProperties: The instruction properties for the specified instruction tuple
+    pub fn instruction_properties(&self, index: usize) -> PyResult<Option<InstructionProperties>> {
+        let mut index_counter = 0;
+        for (_operation, props_map) in self.gate_map.iter() {
+            let gate_map_oper = props_map.values();
+            for inst_props in gate_map_oper {
+                if index_counter == index {
+                    return Ok(inst_props.clone());
+                }
+                index_counter += 1;
+            }
+        }
+        Err(PyIndexError::new_err(format!(
+            "Index: {:?} is out of range.",
+            index
+        )))
+    }
+
+    /// Return the non-global operation names for the target
+    ///
+    /// The non-global operations are those in the target which don't apply
+    /// on all qubits (for single qubit operations) or all multi-qubit qargs
+    /// (for multi-qubit operations).
+    ///
+    /// Args:
+    ///     strict_direction (bool): If set to ``True`` the multi-qubit
+    ///         operations considered as non-global respect the strict
+    ///         direction (or order of qubits in the qargs is significant). For
+    ///         example, if ``cx`` is defined on ``(0, 1)`` and ``ecr`` is
+    ///         defined over ``(1, 0)`` by default neither would be considered
+    ///         non-global, but if ``strict_direction`` is set ``True`` both
+    ///         ``cx`` and ``ecr`` would be returned.
+    ///
+    /// Returns:
+    ///     List[str]: A list of operation names for operations that aren't global in this target
+    #[pyo3(name = "get_non_global_operation_names", signature = (/, strict_direction=false,))]
+    fn py_get_non_global_operation_names(
+        &mut self,
+        py: Python<'_>,
+        strict_direction: bool,
+    ) -> PyObject {
+        self.get_non_global_operation_names(strict_direction)
+            .to_object(py)
+    }
+
+    // Class properties
```

Comment:
```suggestion
    // Instance attributes
```

or `properties` are fine too. But they're not class properties because they're defined for each object and not at the class level.